src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -443,6 +446,20 @@ private void PreInitialize()
                 }
             }
 
+            if (string.IsNullOrEmpty(_environment.GetFunctionsWorkerRuntime()))
+            {
+                string baseMessage = $"The '{EnvironmentSettingNames.FunctionWorkerRuntime}' App Setting is required. Please specify a valid value. See {DiagnosticEventConstants.MissingFunctionsWorkerRuntimeHelpLink} for more information.";
+                if (_hostingConfigOptions.Value.ThrowOnMissingFunctionsWorkerRuntime)
+                {
+                    _logger.LogDiagnosticEventError(DiagnosticEventConstants.MissingFunctionsWorkerRuntimeErrorCode, baseMessage, DiagnosticEventConstants.MissingFunctionsWorkerRuntimeHelpLink, null);
+                    throw new HostInitializationException(baseMessage);
+                }
+
+                string warningMessage = baseMessage + " The app will continue to run, but may throw an exception in a future release.";
+                _logger.LogDiagnosticEventError(DiagnosticEventConstants.MissingFunctionsWorkerRuntimeErrorCode, warningMessage + " Please .", DiagnosticEventConstants.MissingFunctionsWorkerRuntimeHelpLink, null);
```

Comment:
Should we (create and) use `LogDiagnosticEventError` here instead of error ?

Reply:
I've updated this to use `LogDiagnosticEventWarning`.